### DISCUSSION TITLE
Getting latest releases

### DISCUSSION THREAD

**Discussion by andreyorst** [clj-kondo/clj-kondo#1348] [Q&A] [answered]
Hi, I'd like to ask if there's a way to get the latest builds of clj-kondo programmatically?

I'm subscribed to the notifications, so I know when newer version is released, but the process of getting it is a bit tedious, meaning that I need to either go to the release page from email, which I often read from phone, or visit GitHub manually.

Ideally I would like to write a small script so I could update it similarly to how `clojure-lsp` project is updated. The problem is, because builds are named after the date of the release there's no predictable way to construct a URL to the latest release, even when using GitHub's feature to redirect from https://github.com/clj-kondo/clj-kondo/releases/latest to the latest released tag.

Clojure-lsp builds, for example, are always named `clojure-lsp-native-static-linux-amd64.zip` so the URL is predictable, and thus can be computed programmatically, given that we can detect the system we're on, and processor's architecture. This is pretty convenient, and I would like to know if there's a particlar reason clj-kondo builds are named as `clj-kondo-2021.08.03-linux-amd64.zip` and if there's a way to automate process of updating from user end.

Thanks!

**Comment by borkdude** [comment, marked as answer]
You could look at the clj-kondo install script how to do this. https://github.com/clj-kondo/clj-kondo/blob/master/script/install-clj-kondo

**Reply by andreyorst** [reply]
ah, cool! It didn't occurred to me that there was an install script all along. I've looked how it's done there, and while this can work for my case I still have a question about the choice of the assets naming.

Storing the version in the repository and then fetching it via `curl` is a neat trick. However wouldn't it be simpler to just call release assets without version included in the name? This would also prevent from forgetting to bump the version in the resources file (unless it's automated, of course).

The only downside I see right now is that users will not be able to figure out what version they have in their downloads directory, without running an executable from that directory/zip. Or are there other concerns? Without  version in the name both the installation script and update functions I use in Emacs can become a bit simpler.

**Reply by borkdude** [reply]
I prefer not to change this unless there is a more compelling reason. It might break existing tooling if I do.
